decoders/netflow/netflow.go

```diff
@@ -117,6 +118,38 @@ func DecodeTemplateSet(payload *bytes.Buffer) ([]TemplateRecord, error) {
 	return records, nil
 }
 
+func DecodeIPFIXTemplateSet(payload *bytes.Buffer) ([]TemplateRecord, error) {
+	records := make([]TemplateRecord, 0)
+	var err error
+	for payload.Len() >= 4 {
+		templateRecord := TemplateRecord{}
+		err = utils.BinaryDecoder(payload, &templateRecord.TemplateId, &templateRecord.FieldCount)
+		if err != nil {
+			break
+		}
+
+		if int(templateRecord.FieldCount) < 0 {
+			return records, NewErrorDecodingNetFlow("Error decoding TemplateSet: zero count.")
+		}
+
+		fields := make([]Field, int(templateRecord.FieldCount))
+		for i := 0; i < int(templateRecord.FieldCount); i++ {
+			field := Field{}
+			err = utils.BinaryDecoder(payload, &field)
+			if field.Type > math.MaxInt16 {
+				field.Type = field.Type - math.MaxInt16
+				// Skip Enterprise Number field - rfc7011#section-3.2
+				_ = payload.Next(4)
```

Comment:
Skipping the 4 bytes fixes the parsing, but I think it needs to track that this was a PEN field inside the `Field` type.  In the data that I have the resulting `Type` will be 1 or 2. The later bits will then just treat those fields as as IPFIX_FIELD_octetDeltaCount or IPFIX_FIELD_packetDeltaCount which would be wrong.